libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularPermissionItem.tsx

```diff
@@ -18,50 +18,71 @@ interface IGranularPermissionItemProps {
     handleSetSelectedPermission: (permission: AccessGranularPermission) => void;
 }
 
-const GranularPermissionSelectItem: React.FC<IGranularPermissionItemProps> = ({
-    permission,
-    grantee,
-    selectedPermission,
-    toggleDropdown,
-    handleSetSelectedPermission,
-    onChange,
-}) => {
-    const intl = useIntl();
+const GranularPermissionSelectItem = forwardRef<HTMLDivElement, IGranularPermissionItemProps>(
+    (
+        { permission, grantee, selectedPermission, toggleDropdown, handleSetSelectedPermission, onChange },
+        ref,
+    ) => {
+        const intl = useIntl();
 
-    const handleOnChange = useCallback(
-        (permission: IGranularPermissionTypeItem) => {
-            toggleDropdown();
-            if (permission.id !== selectedPermission) {
-                handleSetSelectedPermission(permission.id);
-                onChange({ ...grantee, permissions: [permission.id] });
-            }
-        },
-        [grantee, toggleDropdown, handleSetSelectedPermission, onChange, selectedPermission],
-    );
+        const handleOnChange = useCallback(
+            (permission: IGranularPermissionTypeItem) => {
+                toggleDropdown();
+                if (permission.id !== selectedPermission) {
+                    handleSetSelectedPermission(permission.id);
+                    onChange({ ...grantee, permissions: [permission.id] });
+                }
+            },
+            [grantee, toggleDropdown, handleSetSelectedPermission, onChange, selectedPermission],
+        );
 
-    const isSelected = useMemo(() => permission.id === selectedPermission, [permission, selectedPermission]);
+        const isSelected = useMemo(
+            () => permission.id === selectedPermission,
+            [permission, selectedPermission],
+        );
 
-    return (
-        <div
-            onClick={() => {
-                if (permission.enabled) {
-                    handleOnChange(permission);
+        const handleKeyDown = useCallback(
+            (e: React.KeyboardEvent) => {
+                if (e.key === "Enter" || e.key === " ") {
+                    e.preventDefault();
+                    if (permission.enabled) {
+                        handleOnChange(permission);
+                    }
                 }
-            }}
-            className={cx(
-                "gd-list-item",
-                "gd-menu-item",
-                "gd-granular-permission-select-item",
-                "s-granular-permission-item",
-                {
-                    "is-disabled": !permission.enabled,
-                    "is-selected": isSelected,
-                },
-            )}
-        >
-            <div>{intl.formatMessage(granularPermissionMessageLabels[permission.id])}</div>
-        </div>
-    );
-};
+            },
+            [permission, handleOnChange],
+        );
+
+        return (
+            <div
```

Comment:
can we make it button to avoid the need for two similar handlers onClick + handleKeyDown?